tests/cross/src/test/scala/tests/pc/AutoImportsSuite.scala

```diff
@@ -504,4 +504,21 @@ class AutoImportsSuite extends BaseAutoImportsSuite {
           |""".stripMargin
     )
 
+  // see: https://github.com/scalameta/metals/issues/8061
+  test("boundary-break".tag(IgnoreScala2)) {
```

Comment:
Are you able to provide a test for Scala 2? The implementation you added is only for Scala 2. The implementation for Scala 3 is in https://github.com/scala/scala3/blob/main/presentation-compiler/src/main/dotty/tools/pc/AutoImportsProvider.scala